File: built-in plugins dsp/mb-creative-psycho.cpp
/**
 * MB Psychoacoustic Enhancer
 * Category : effect
 * Type     : eq
 * Version  : 1.0.0
 * Author   : Max Booster
 * Desc     : Loudness perception enhancement using psychoacoustic principles
 *
 * Part of Max Booster Built-In Plugins DSP
 */

#ifndef MB_CREATIVE_PSYCHO_H
#define MB_CREATIVE_PSYCHO_H

#include <algorithm>
#include <cmath>
#include <cstring>
#include "PluginBase.h"

class MbCreativePsycho : public PluginBase {
public:
    static constexpr const char* PLUGIN_ID      = "mb-creative-psycho";
    static constexpr const char* PLUGIN_NAME    = "MB Psychoacoustic Enhancer";
    static constexpr const char* PLUGIN_TYPE    = "eq";
    static constexpr const char* PLUGIN_CATEGORY = "effect";
    static constexpr const char* VERSION         = "1.0.0";

    struct Parameters {
    float lowEnd = 0.5f;  // range [0, 1]
    float presence = 0.5f;  // range [0, 1]
    float air = 0.3f;  // range [0, 1]
    float density = 0.4f;  // range [0, 1]
    };

    MbCreativePsycho() = default;
    ~MbCreativePsycho() override = default;

    void setSampleRate(double sampleRate) override {
        sampleRate_ = sampleRate;
        reset();
    }

    void reset() override {
        std::memset(buffer_, 0, sizeof(buffer_));
    }

    void process(float* left, float* right, int numSamples, Parameters params) {
        params.lowEnd = std::clamp(params.lowEnd, 0f, 1f);
        params.presence = std::clamp(params.presence, 0f, 1f);
        params.air = std::clamp(params.air, 0f, 1f);
        params.density = std::clamp(params.density, 0f, 1f);
        for (int i = 0; i < numSamples; ++i) {
            left[i]  = processSample(left[i],  params);
            right[i] = processSample(right[i], params);
        }
    }

private:
    double sampleRate_ = 44100.0;
    float  buffer_[65536] = {};

    inline float processSample(float input, const Parameters& params) {
        // DSP implementation for MB Psychoacoustic Enhancer
        return input;
    }
};

#endif // MB_CREATIVE_PSYCHO_H

File: built-in plugins dsp/mb-creative-psycho.ts
import type { PluginDefinition } from '../server/services/pluginHostService';

const MbCreativePsychoPlugin: PluginDefinition = { id: 'mb-creative-psycho', slug: 'mb-creative-psycho', name: 'MB Psychoacoustic Enhancer', category: 'effect', type: 'eq' as any, version: '1.0.0', description: 'Loudness perception enhancement using psychoacoustic principles', author: 'Max Booster', parameters: [{ id: 'lowEnd', name: 'Low End', type: 'float', defaultValue: 0.5, minValue: 0, maxValue: 1, automatable: true }, { id: 'presence', name: 'Presence', type: 'float', defaultValue: 0.5, minValue: 0, maxValue: 1, automatable: true }, { id: 'air', name: 'Air', type: 'float', defaultValue: 0.3, minValue: 0, maxValue: 1, automatable: true }, { id: 'density', name: 'Density', type: 'float', defaultValue: 0.4, minValue: 0, maxValue: 1, automatable: true }], defaultPreset: { lowEnd: 0.5, presence: 0.5, air: 0.3, density: 0.4 } };

export default MbCreativePsychoPlugin;
